app/src/main/java/com/example/hellobuttons/CounterActivity.kt

```diff
@@ -0,0 +1,27 @@
+package com.example.hellobuttons
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.TextView
+
+class CounterActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_counter)
+
+        // Get the Intent that started this activity and extract the value
+
+        // option 1
+        val bundle: Bundle? = intent.extras      // built in?
```

Comment:
so is intent a built in property of any activity?

Reply:
Yes - every activity has to have an Intent to start. The bundle isn't always present, but the intent is.

Reply:
cool, can it be empty? like does the main activity have one?

Reply:
the intent is never null - unless something bad happens. The bundle can be null, absolutely. 

Even the main activity can have bundle args, this is very common - think deep links

Reply:
got it. thanks